test/unit/lib/plugins/aws/invoke.test.js

```diff
@@ -34,6 +36,95 @@ describe('AwsInvoke', () => {
     awsInvoke = new AwsInvoke(serverless, options);
   });
 
+  it('should allow invoke with context', async () => {
+    const lambdaInvokeStub = sinon.stub();
+
+    const result = await runServerless({
+      fixture: 'apiGateway',
+      cliArgs: ['invoke', '-f', 'foo', '--context', 'somecontext'],
+      awsRequestStubMap: {
+        Lambda: {
+          invoke: args => {
+            lambdaInvokeStub.returns('payload');
+            return lambdaInvokeStub(args);
+          },
+        },
+      },
+    });
+    expect(lambdaInvokeStub).to.have.been.calledOnce;
+    expect(lambdaInvokeStub).to.have.been.calledWithExactly({
+      ClientContext: 'c29tZWNvbnRleHQ=',
+      FunctionName: result.serverless.service.getFunction('foo').name,
+      InvocationType: 'RequestResponse',
+      LogType: 'None',
+      Payload: sinon.match({}),
+    });
+  });
+
+  ['-x', '--contextPath'].forEach(option => {
+    it(`should allow invoke with contextPath for option ${option}`, async () => {
+      const lambdaInvokeStub = sinon.stub();
+
+      serverless.config.servicePath = getTmpDirPath();
+      const data = {
+        testProp: 'testValue',
+      };
+      const contextDataFile = path.join(serverless.config.servicePath, 'context.json');
+      serverless.utils.writeFileSync(contextDataFile, JSON.stringify(data));
+
+      const result = await runServerless({
+        fixture: 'apiGateway',
+        cliArgs: ['invoke', '-f', 'foo', option, contextDataFile],
+        awsRequestStubMap: {
+          Lambda: {
+            invoke: args => {
+              lambdaInvokeStub.returns('payload');
+              return lambdaInvokeStub(args);
+            },
+          },
+        },
+      });
+      expect(lambdaInvokeStub).to.have.been.calledOnce;
+      expect(lambdaInvokeStub).to.have.been.calledWithExactly({
+        ClientContext: 'eyJ0ZXN0UHJvcCI6InRlc3RWYWx1ZSJ9',
+        FunctionName: result.serverless.service.getFunction('foo').name,
+        InvocationType: 'RequestResponse',
+        LogType: 'None',
+        Payload: sinon.match({}),
```

Comment:
Does this work with `chai.expect` ? 

Alternatively, we can probably use `args[0][0]` and `deep.equals` (?)